contracts/proxies/ProxyFactory.sol

```diff
@@ -56,4 +56,26 @@ contract ProxyFactory {
             }
         emit ProxyCreation(proxy);
     }
+
+    /// @dev Allows to get the address for a new proxy contact created via `createProxyWithNonce`
+    ///      This method is only meant for address calculation purpose, therefore the response is returned with a revert
+    ///      When calling this method set `from` to the address of the proxy.
+    /// @param _mastercopy Address of master copy.
+    /// @param initializer Payload for message call sent to new proxy contract.
+    /// @param saltNonce Nonce that will be used to generate the salt to calculate the address of the new proxy contract.
+    function calculateCreateProxyWithNonceAddress(address _mastercopy, bytes calldata initializer, uint256 saltNonce)
+        external
+        returns (Proxy proxy)
+    {
+        // If the initializer changes the proxy address should change too. Hashing the initializer data is cheaper than just concatinating it
+        bytes32 salt = keccak256(abi.encodePacked(keccak256(initializer), saltNonce));
```

Comment:
Should we make this into a method (since these couple lines are used by both methods). It will costs a little bit more gas, but will make it more readable. (Maybe we should test how much more gas it would be actually)

Reply:
@rmeissner Currently, gas costs are:
`factory.createProxyWithNonce('0x0000000000000000000000000000000000000000', '0x00', '0x00')`
gasUsed: 58909,
cumulativeGasUsed: 58909

If I extract create2 to one method, it will be:
`factory.createProxyWithNonce('0x0000000000000000000000000000000000000000', '0x00', '0x00')`
gasUsed: 58962
cumulativeGasUsed: 58962

I guess with that little gas increase we can use the function